Vladimir-ReactJS-w1

intruston opened the pull request.

intruston commented:
Here's the link for deployed app https://golden-piroshki-078b4d.netlify.app/

crevulus commented on week1/project/ecommerce/src/App.js:
```diff
@@ -0,0 +1,59 @@
+
+  return (
+    <>
+      <h1>Products</h1>
```
The beauty of React and other frameworks is that you can sort your UI elements into components, rather than putting evrything into one file. Try breaking this JSX down into components. https://beta.reactjs.org/learn/your-first-component 🔴

crevulus commented on week1/project/ecommerce/src/index.css:
```diff
@@ -0,0 +1,246 @@
+  overflow: visible
+}
+
+pre {
```
There's a whole lot of styles in here for elements that you do not use. If you are not using these styles then it's best to remove them. 🟠

crevulus commented on week1/project/ecommerce/src/fake-data/all-categories.js:
```diff
@@ -0,0 +1,6 @@
+export default [
```
You have a couple of warnings in your terminal. They are not critical, but ideally you should have 0 warnings or errors. 🟢

crevulus commented on week1/project/ecommerce/src/App.js:
```diff
@@ -0,0 +1,59 @@
+          </div>
+        ))}
+      </div>
+      <ul class="products">
```
`class` can be used in HTML, but not in JSX. Check your browser terminal for an error. 🔴

crevulus commented on week1/project/ecommerce/src/App.js:
```diff
@@ -0,0 +1,59 @@
+      <div className="categories">
+        {categories.map((category) => (
+          <div
+            className={`categories--item${
```
Nice interpolation and ternary operator!

crevulus commented on week1/project/ecommerce/src/App.js:
```diff
@@ -0,0 +1,59 @@
+      <h1>Products</h1>
+      <div className="categories">
+        {categories.map((category) => (
+          <div
```
This should not be a `div` -- be careful to use semantic html in your projects, otherwise you will make your website harder to use. https://www.semrush.com/blog/semantic-html5-guide/ 🟠

crevulus reviewed (commented):
Hi Vladimir, great work on this app. The only core React concept that you've missed is the power of Components - try using them to your advantage! Other than than, this is a very strong first React app.

intruston commented:
Hi @crevulus Christopher! Thank you for review. I tried to split my app to Components and added CategoryAll and ProductList. Also fixed classes and css 👍
